compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -2305,6 +2307,16 @@ public void visit(BLangConstant constant) {
                 constantSymbol.type.tsymbol.flags |= constant.associatedTypeDefinition.symbol.flags;
             }
 
+        } else if (nodeKind == NodeKind.UNARY_EXPR && constant.typeNode == null &&
```

Comment:
Why are special casing at both node builder and here? Please see the comment for the node builder also.

Reply:
Same reason as[ [1]](https://github.com/ballerina-platform/ballerina-lang/pull/34622#discussion_r871193315)